static/app/components/modals/sudoModal.tsx

```diff
@@ -28,6 +28,7 @@ type Props = WithRouterProps &
      * User is a superuser without an active su session
      */
     isSuperuser?: boolean;
+    needsReload?: boolean;
```

Comment:
Shouldn't sudo maybe just _always_ reload? Why is it just in this one place?

Is there maybe a way we can totally avoid a reload and re-init some client context?

Reply:
The purpose of this is for the flow where you access a customer's org but don't have superuser access. In this case, even after being granted superuser, you still will be on the same page with nothing loaded and an error message so a page refresh is probably the best way for it to load everything that you need.

In every other case of opening the superuse/sudo modal, you don't need to reload anything as the purpose is to be given superuser/sudo access and closing the modal after success is good

Reply:
Is there any case where they open the modal and are _not_ granted superuser access?

What would happen in that case after closing the modal?

Reply:
That would only happen if the API request failed. That will be handled by `handleError` while this is in `handleSuccess`